pkg/syncutil/stats_reporter.go

```diff
@@ -181,26 +145,46 @@ func NewStatsReporter() (*Reporter, error) {
 	return &Reporter{now: now}, nil
 }
 
+func (r *Reporter) RegisterCallback(c *MetricsCache) error {
+	_, err1 := meter.RegisterCallback(c.ReportSync, syncM)
+	_, err2 := meter.RegisterCallback(r.ReportLastSync, lastRunSyncM)
+	return errors.Join(err1, err2)
+}
+
 func (r *Reporter) ReportSyncDuration(d time.Duration) error {
 	ctx := context.Background()
-	return metrics.Record(ctx, syncDurationM.M(d.Seconds()))
+	syncDurationM.Record(ctx, d.Seconds())
+	return nil
 }
 
-func (r *Reporter) ReportLastSync() error {
-	ctx := context.Background()
-	return metrics.Record(ctx, lastRunSyncM.M(r.now()))
+func (r *Reporter) ReportLastSync(_ context.Context, observer metric.Observer) error {
```

Comment:
The `ReportX` functions should still be used for reporting, not observing.

In general `ReportX` functions:

* Update metric state directly, where open telemetry allows for it
* If open telemetry does not allow for it (seems to be where `observeX` functions are used), they update internal reporter variables, that are observed by `observeX` functions.

Concurrency also needs to be taken into account when updating internal state (atomic variables and/or mutexes)